core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -153,7 +153,7 @@ public CarbonTable getTable() {
     int carbonDriverPruningMultiThreadEnableFilesCount =
         CarbonProperties.getDriverPruningMultiThreadEnableFilesCount();
     if (numOfThreadsForPruning == 1 || indexesCount < numOfThreadsForPruning || totalFiles
-            < carbonDriverPruningMultiThreadEnableFilesCount) {
+            < carbonDriverPruningMultiThreadEnableFilesCount || !isFilterPresent) {
```

Comment:
here add a comment saying, when the query is without filter, as we need to return all the blocklets, no need to prune multithread

Reply:
added.